modules/java/native/java-grammar.ym

```diff
@@ -86,6 +87,14 @@ java_dest_custom_options
   |
   ;
 
+java_dest_empty_options
+  : java_dest_empty_option java_dest_empty_options
+  |
+  ;
+
+java_dest_empty_option
+  : LL_STRING { ;  }
+
```

Comment:
Do you have an example? I may miss some of the larger picture.
thanks.

-- 
Bazsi

On Mon, Jul 13, 2015 at 2:42 PM, aneutrals <notifications@github.com> wrote:

> In modules/java/native/java-grammar.ym
> <https://github.com/balabit/syslog-ng/pull/556#discussion_r34458564>:
>
> > @@ -86,6 +87,14 @@ java_dest_custom_options
> >    |
> >    ;
> >
> > +java_dest_empty_options
> > +  : java_dest_empty_option java_dest_empty_options
> > +  |
> > +  ;
> > +
> > +java_dest_empty_option
> > +  : LL_STRING { ;  }
> > +
>
> Just a note, bailing out with an error message would unfix the issue,
> because it would still force destinations to always be passed values for
> every possible optional parameter, behavior that might not be acceptable
> for some destinations.
>
> That being said, as an alternative to this patch, we could simply say that
> all destinations must assume all of their optional parameters will always
> get values, even if the values are defaults defined in the SCL blocks. I
> don't know if this would be limiting though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/556/files#r34458564>.
>